R/zchunk_LA1231.elec_tech.R

```diff
@@ -37,59 +38,165 @@ module_energy_LA1231.elec_tech_DISABLED <- function(command, ...) {
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # Natural gas: Disaggregate to CC and CT/steam on the basis of assumed efficiencies
+    # Subset the actual efficiencies of gas -> electricity
+    L123.eff_R_elec_F_Yh %>%
+      filter(fuel == "gas") %>%
+      rename(efficiency = value) -> L1231.eff_R_elec_gas_Yh_gathered
+
+    # Create auxiliar tibbles to perform interpolation (aprox_fun) on efficiencies for natural gas technologies
+    # Interpolation is needed since input data is only provided for some intermediate years in the range 1971-2010.
+    # Therefore there is not efficiency values for every historical year. Efficiencies for all historical years are needed
+    # To estimate outputs from fuel/technology by region in the electricity sector
+    tibble(technology = energy.GAS_TECH, supplysector = "electricity") -> Gas_tech_elec
+    tibble(supplysector = "electricity", subsector="gas", year = HISTORICAL_YEARS) %>%
+      mutate(year = as.numeric(year)) %>%
+      full_join(Gas_tech_elec, by = "supplysector") -> Aux_gas_tech_elec
+
+    # Perform interpolation for gas technologies efficiencies , max, and rate
+    # Interpolation is needed since efficiencies are given for 1971, 1990, 2005 and 2010 only. Not for every historical year.
+    A23.globaltech_eff %>%
+      filter(subsector == "gas") %>%
+      semi_join(calibrated_techs, by = c("supplysector", "subsector", "technology")) %>%
+      gather(year, efficiency_tech, -supplysector, -subsector, -technology, -minicam.energy.input, -improvement.max, -improvement.rate, -improvement.shadow.technology) %>%
+      mutate(year = as.numeric(year)) %>%
+      mutate(efficiency_tech = as.numeric(efficiency_tech)) %>%
+      semi_join(Aux_gas_tech_elec, by = "year") %>%
+      full_join(Aux_gas_tech_elec, by = c("supplysector", "subsector", "technology","year")) %>%
+      group_by(supplysector, subsector, technology) %>%
+      mutate(efficiency_tech = approx_fun(year, efficiency_tech)) %>%
+      mutate(improvement.max = approx_fun(year, improvement.max)) %>%
+      mutate(improvement.rate = approx_fun(year, improvement.rate)) %>%
+      mutate(minicam.energy.input = "wholesale gas") -> L1231.eff_R_elec_gas_tech
+
```

Comment:
Again the fuel name, "wholesale gas", must come from calibrated_tech.csv or A23.globaltech_eff.csv so that fuel inputs can be changed through data rather than code.